### ISSUE TITLE
Selinux problems

### ISSUE THREAD

**Issue by janinko** [release-engineering/koji-dojo#19]
Fedora 24
I have selinux problems while running build scripts:

```
SELinux is preventing openssl from write access on the file uptime.

*****  Plugin catchall (100. confidence) suggests   **************************

If you believe that openssl should be allowed write access on the uptime file by default.
Then you should report this as a bug.
You can generate a local policy module to allow this access.
Do
allow this access for now by executing:
# ausearch -c 'openssl' --raw | audit2allow -M my-openssl
# semodule -X 300 -i my-openssl.pp

Additional Information:
Source Context                system_u:system_r:svirt_lxc_net_t:s0:c693,c940
Target Context                system_u:object_r:proc_t:s0
Target Objects                uptime [ file ]
Source                        openssl
Source Path                   openssl
Port                          <Unknown>
Host                          janinko-redhat
Source RPM Packages           
Target RPM Packages           
Policy RPM                    selinux-policy-3.13.1-191.14.fc24.noarch
Selinux Enabled               True
Policy Type                   targeted
Enforcing Mode                Enforcing
Host Name                     janinko-redhat
Platform                      Linux janinko-redhat 4.6.5-300.fc24.x86_64 #1 SMP
                              Thu Jul 28 01:10:12 UTC 2016 x86_64 x86_64
Alert Count                   6
First Seen                    2016-09-19 14:25:00 CEST
Last Seen                     2016-09-19 14:25:00 CEST
Local ID                      93a16571-3083-436e-a07a-42c84ec293c6

Raw Audit Messages
type=AVC msg=audit(1474287900.477:352): avc:  denied  { write } for  pid=8382 comm="openssl" name="uptime" dev="proc" ino=4026531992 scontext=system_u:system_r:svirt_lxc_net_t:s0:c693,c940 tcontext=system_u:object_r:proc_t:s0 tclass=file permissive=0


Hash: openssl,svirt_lxc_net_t,proc_t,file,write
```

**Comment by janinko** [comment]
When starting the hub I got these problems:
```
SELinux is preventing git from write access on the directory koji-clients.

*****  Plugin catchall_labels (83.8 confidence) suggests   *******************

If you want to allow git to have write access on the koji-clients directory
Then you need to change the label on koji-clients
Do
# semanage fcontext -a -t FILE_TYPE 'koji-clients'
where FILE_TYPE is one of the following: cgroup_t, docker_var_lib_t, nfs_t, svirt_home_t, svirt_sandbox_file_t, virt_home_t.
Then execute:
restorecon -v 'koji-clients'


*****  Plugin catchall (17.1 confidence) suggests   **************************

If you believe that git should be allowed write access on the koji-clients directory by default.
Then you should report this as a bug.
You can generate a local policy module to allow this access.
Do
allow this access for now by executing:
# ausearch -c 'git' --raw | audit2allow -M my-git
# semodule -X 300 -i my-git.pp

Additional Information:
Source Context                system_u:system_r:svirt_lxc_net_t:s0:c817,c917
Target Context                unconfined_u:object_r:usr_t:s0
Target Objects                koji-clients [ dir ]
Source                        git
Source Path                   git
Port                          <Unknown>
Host                          janinko-redhat
Source RPM Packages           
Target RPM Packages           
Policy RPM                    selinux-policy-3.13.1-191.14.fc24.noarch
Selinux Enabled               True
Policy Type                   targeted
Enforcing Mode                Enforcing
Host Name                     janinko-redhat
Platform                      Linux janinko-redhat 4.6.5-300.fc24.x86_64 #1 SMP
                              Thu Jul 28 01:10:12 UTC 2016 x86_64 x86_64
Alert Count                   6
First Seen                    2016-09-19 14:33:26 CEST
Last Seen                     2016-09-19 14:33:40 CEST
Local ID                      35a50526-b76b-44f0-aa01-fd70a74de219

Raw Audit Messages
type=AVC msg=audit(1474288420.49:437): avc:  denied  { write } for  pid=15759 comm="mkdir" name="koji-clients" dev="dm-1" ino=1441798 scontext=system_u:system_r:svirt_lxc_net_t:s0:c817,c917 tcontext=unconfined_u:object_r:usr_t:s0 tclass=dir permissive=0


Hash: git,svirt_lxc_net_t,usr_t,dir,write
```
```
SELinux is preventing chown from setattr access on the directory koji-clients.

*****  Plugin catchall_labels (83.8 confidence) suggests   *******************

If you want to allow chown to have setattr access on the koji-clients directory
Then you need to change the label on koji-clients
Do
# semanage fcontext -a -t FILE_TYPE 'koji-clients'
where FILE_TYPE is one of the following: cgroup_t, nfs_t, svirt_home_t, svirt_sandbox_file_t.
Then execute:
restorecon -v 'koji-clients'


*****  Plugin catchall (17.1 confidence) suggests   **************************

If you believe that chown should be allowed setattr access on the koji-clients directory by default.
Then you should report this as a bug.
You can generate a local policy module to allow this access.
Do
allow this access for now by executing:
# ausearch -c 'chown' --raw | audit2allow -M my-chown
# semodule -X 300 -i my-chown.pp

Additional Information:
Source Context                system_u:system_r:svirt_lxc_net_t:s0:c817,c917
Target Context                unconfined_u:object_r:usr_t:s0
Target Objects                koji-clients [ dir ]
Source                        chown
Source Path                   chown
Port                          <Unknown>
Host                          janinko-redhat
Source RPM Packages           
Target RPM Packages           
Policy RPM                    selinux-policy-3.13.1-191.14.fc24.noarch
Selinux Enabled               True
Policy Type                   targeted
Enforcing Mode                Enforcing
Host Name                     janinko-redhat
Platform                      Linux janinko-redhat 4.6.5-300.fc24.x86_64 #1 SMP
                              Thu Jul 28 01:10:12 UTC 2016 x86_64 x86_64
Alert Count                   1
First Seen                    2016-09-19 14:33:40 CEST
Last Seen                     2016-09-19 14:33:40 CEST
Local ID                      dc7ba37a-501d-429b-9b51-751c9fc67b0c

Raw Audit Messages
type=AVC msg=audit(1474288420.56:438): avc:  denied  { setattr } for  pid=15768 comm="chown" name="koji-clients" dev="dm-1" ino=1441798 scontext=system_u:system_r:svirt_lxc_net_t:s0:c817,c917 tcontext=unconfined_u:object_r:usr_t:s0 tclass=dir permissive=0


Hash: chown,svirt_lxc_net_t,usr_t,dir,setattr
```

**Comment by janinko** [comment]
To solve the last two problems do:
```bash
 sudo semanage fcontext -a -t svirt_sandbox_file_t /opt/koji
 sudo semanage fcontext -a -t svirt_sandbox_file_t /opt/koji-clients
 sudo semanage fcontext -a -t svirt_sandbox_file_t /opt/koji-files

sudo restorecon -vr /opt/koji*
```
That will set proper label for the folders so the docker has acces to it. This could be added to the README file. More info: https://www.projectatomic.io/blog/2015/06/using-volumes-with-docker-can-cause-problems-with-selinux/

**Comment by jdcasey** [comment]
Hi @janinko, are you using volume mounts, maybe? If so, you might need to execute something like this:

```
$ chcon -Rt svirt_sandbox_file_t <dir>
```

Just to let selinux know those directories are usable from a container like Docker. I've had to do this a few times in CentOS, but I haven't tried running koji-dojo (or anything else, really) on Fedora.

**Comment by janinko** [comment]
Hi, yes, something like that helped with the second selinux problems. This info could go into the readme.

But I'm not sure about the first problem:
```
SELinux is preventing openssl from write access on the file uptime.
```

**Comment by jwmullally** [comment]
Under Fedora 25, "svirt_sandbox_file_t" appears to has been removed/renamed. "container_file_t" worked for me. The above patch should handle this automatically.

**Comment by jdcasey** [comment]
Merged PR#24 which contained the above patch.
